File: hackathon_CPP_SDK/src-gen/nevonex/types/PropertyChange.cpp
/*
 * nevonex/types/PropertyChange.cpp
 * Copyright (c) Robert Bosch GmbH. All rights reserved.
 */

#include "PropertyChange.hpp"
#include <algorithm>

using namespace ::nevonex::types;

// Default constructor
PropertyChange::PropertyChange()
{
}

PropertyChange::~PropertyChange()
{
}

// Attributes

std::vector< ::nevonex::types::PropertyChangeListener > const& PropertyChange::getListeners() const
{
    return m_listeners;
}

void PropertyChange::addListeners(::nevonex::types::PropertyChangeListener _new_element_in_listeners)
{
    m_listeners.push_back(_new_element_in_listeners);
}

void PropertyChange::setListenersAt(size_t _position, ::nevonex::types::PropertyChangeListener _new_element_in_listeners)
{
    if (_position >= m_listeners.size()) m_listeners.resize(_position + 1, 0);
    m_listeners[_position] = _new_element_in_listeners;
}

void PropertyChange::deleteListenersAt(size_t _position)
{
    if (_position >= m_listeners.size()) return;
    m_listeners.erase(m_listeners.begin() + _position);
}

void PropertyChange::removeListeners(::nevonex::types::PropertyChangeListener _remove)
{
    m_listeners.erase(std::remove(m_listeners.begin(), m_listeners.end(), _remove), m_listeners.end());
}

// Framework operations

void PropertyChange::addPropertyChangeListener(::nevonex::types::PropertyChangeListener _listener)
{
    addListeners(_listener);
}

void PropertyChange::removePropertyChangeListener(::nevonex::types::PropertyChangeListener _listener)
{
    removeListeners(_listener);
}

void PropertyChange::notifyPropertyChange(::ecore::EString const& _name, ::ecore::EJavaObject const& _oldValue, ::ecore::EJavaObject const& _newValue)
{
    /*PROTECTED REGION ID(PropertyChange_notifyPropertyChange) START*/
    /*PROTECTED REGION END*/
}

File: com.bosch.fsp.hackathon.gen/src/com/bosch/nevonex/types/impl/PropertyChange.java
/**
Copyright (c) Robert Bosch GmbH. All rights reserved.
*/
package com.bosch.nevonex.types.impl;

import com.bosch.nevonex.types.IPropertyChange;
import com.bosch.nevonex.types.impl.TypesPackage;

import java.beans.PropertyChangeEvent;
import java.beans.PropertyChangeListener;

import org.eclipse.emf.common.util.BasicEList;
import org.eclipse.emf.common.util.EList;

import org.eclipse.emf.ecore.EClass;
import org.eclipse.emf.ecore.impl.EObjectImpl;

/**
 * An implementation of the model object '<em><b>PropertyChange</b></em>'.
 * @generated
 */
public class PropertyChange extends EObjectImpl implements IPropertyChange {
	/**
	 * @generated
	 */
	protected EList<PropertyChangeListener> listeners;

	/**
	 * @generated
	 */
	protected PropertyChange() {
		super();
	}

	/**
	 * @generated
	 */
	@Override
	protected EClass eStaticClass() {
		return TypesPackage.Literals.PROPERTY_CHANGE;
	}

	/**
	 * @generated
	 */
	public void addPropertyChangeListener(PropertyChangeListener listener) {
		if (listeners == null) {
			listeners = new BasicEList<>();
		}
		listeners.add(listener);
	}

	/**
	 * @generated
	 */
	public void removePropertyChangeListener(PropertyChangeListener listener) {
		if (listeners != null) {
			listeners.remove(listener);
		}
	}

	/**
	 * @generated
	 */
	public void notifyPropertyChange(String name, Object oldValue, Object newValue) {
		if (listeners != null) {
			for (PropertyChangeListener listener : this.listeners) {
				listener.propertyChange(new PropertyChangeEvent(this, name, oldValue, newValue));
			}
		}
	}
} //PropertyChange
